perfkitbenchmarker/publisher.py

```diff
@@ -736,3 +758,56 @@ def PublishSamples(self):
     for publisher in self.publishers:
       publisher.PublishSamples(self.samples)
     self.samples = []
+
+
+def RepublishJSONSamples(path):
+  """Read samples from a JSON file and re-export them.
+
+  Args:
+    path: the path to the JSON file.
+  """
+
+  samples = []
+  with open(path, 'r') as file:
+    for line in file:
+      try:
+        raw_dict = json.loads(line)
+        labels = raw_dict.pop('labels')
+        metadata = dict()
+        # labels is a comma-seprated list of key, value pairs
+        for pair in labels.split(','):
+          # Need to strip leading and trailing '|' from each pair
+          key, _, value = pair[1:-1].partition(':')
```

Comment:
Very nice! I like your code. I pushed a very small variation on it - just have to pop 'labels' so that there's not an extra field in the result. Thanks a lot!